openscenario/openscenario_interpreter/include/openscenario_interpreter/syntax/condition.hpp

```diff
@@ -49,9 +57,36 @@ struct Condition : public ComplexType
 
   bool current_value;
 
+private:
+  struct History
+  {
+    double time;
+    bool result;
+  };
+
+  std::list<History> histories;
+
+public:
   explicit Condition(const pugi::xml_node & node, Scope & scope);
 
   auto evaluate() -> Object;
+
+private:
+  template <class R, class... Args>
+  auto update_condition(std::function<R(Args...)> condition) -> Object
```

Comment:
This function implicitly requires "a function whose argument types are any number of bool" (or "a function whose argument types are any number of types compatible with bool").
So it's nicer to express the type requirements as template parameter names.

```suggestion
  template <typename... Booleans>
  auto update_condition(std::function<bool(Booleans...)> condition)
```